From: Edward Hervey <edward.hervey@collabora.co.uk>
Subject: baseaudiosink: Resync when ringbuffer resets

When the ringbuffer gets restarted (like in setcaps), we *will* have to 
resync against the new values. Without this we end up blindly assuming the 
new samples align to the old ones.

Origin: upstream, http://cgit.freedesktop.org/gstreamer/gst-plugins-base/commit/?id=969fe8c0b97ae4ac58aacf7530ca1bfbc2b5969a
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=680252
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=668109
Index: gst-plugins-base0.10-0.10.36/gst-libs/gst/audio/gstbaseaudiosink.c
===================================================================
--- gst-plugins-base0.10-0.10.36.orig/gst-libs/gst/audio/gstbaseaudiosink.c	2011-12-30 14:29:13.000000000 +0100
+++ gst-plugins-base0.10-0.10.36/gst-libs/gst/audio/gstbaseaudiosink.c	2012-12-04 11:02:39.294066822 +0100
@@ -921,6 +921,12 @@
   if (!gst_ring_buffer_acquire (sink->ringbuffer, spec))
     goto acquire_error;
 
+  /* We need to resync since the ringbuffer restarted */
+  sink->priv->avg_skew = -1;
+  sink->next_sample = -1;
+  sink->priv->eos_time = -1;
+  sink->priv->discont_time = -1;
+
   if (bsink->pad_mode == GST_ACTIVATE_PUSH) {
     GST_DEBUG_OBJECT (sink, "activate ringbuffer");
     gst_ring_buffer_activate (sink->ringbuffer, TRUE);
